src/rosflight_joy/rc_keyboard

```diff
@@ -4,56 +4,99 @@
 
 import rospy
 from rosflight_msgs.msg import RCRaw, Status
+from std_msgs.msg import String
 from rosflight_keyboard_base import rosflight_keyboard_base
 
 class rosflight_keyboard_RC(rosflight_keyboard_base):
     def __init__(self, device = 0):
         rosflight_keyboard_base.__init__(self, device)
 
         rospy.init_node('rc_keyboard')
-        
-        rc_pub = rospy.Publisher('RC', RCRaw, queue_size=10)
-        
-        rospy.Subscriber('status', Status, self.status_callback)
-        self.auto_arm = rospy.get_param("~auto_arm", False) 
-        self.is_armed = False
-        self.init = False
+        rospy.loginfo("[rc_keyboard] initing")
 
-        rospy.sleep(3)
-        rospy.loginfo("[rc_keyboard] init successful")
-        
-        while not rospy.is_shutdown():
+        self.rc_pub = rospy.Publisher('RC', RCRaw, queue_size=10)
 
-            if not self.init:                                             
-                if self.auto_arm and self.is_armed:                       
-                    rospy.loginfo("[rc_keyboard] auto-arming successful") 
-                    self.init = True                                      
-            
-            self.update()
+        self.status_sub = rospy.Subscriber('status', Status, self.status_callback)
+        # waiting until firmware version is published avoids some "Unable to arm" errors
+        self.version_sub = rospy.Subscriber('version', String, self.version_callback)
+
+        self.init = False
+        self.is_armed = False
+        self.got_version = False
+        self.auto_arm = rospy.get_param('~auto_arm', False)
+        self.auto_takeoff = rospy.get_param('~auto_takeoff', False)
+        if not self.auto_arm:
+            self.auto_takeoff = False
+            self.init = True
+        self.rate = rospy.Rate(30)
+        if self.auto_arm:
+            rospy.logwarn("[rc_keyboard] AUTO-ARM is ON. Waiting for firmware version")
+        else:
+            rospy.loginfo("[rc_keyboard] AUTO-ARM is OFF.")
+        if self.auto_takeoff:
+            rospy.logwarn("[rc_keyboard] AUTO-TAKEOFF is ON.")
+        else:
+            rospy.loginfo("[rc_keyboard] AUTO-TAKEOFF is OFF.")
+    def run(self):
+        while not rospy.is_shutdown():
+            self.rate.sleep()
           
+            if self.got_version:
+                if self.auto_arm and not self.init:
+                    self.arm()
+                    if self.is_armed:
+                        self.values['aux1'] = -1
+                        if self.auto_takeoff:
+                            self.values['F'] = 1
```

Comment:
So will auto takeoff always set full throttle? A `takeoff_throttle` parameter (0 to 1) could be nice to make that less extreme

Reply:
A throttle parameter would be nice. I'm not sure why it was set to full throttle; possibly to ensure it was higher than rosflight's `MIN_THROTTLE`? @davidcGIThub was the one who requested `auto_takeoff`, so he may know why.

Reply:
So I believe the reason it's set to full throttle is because this is the throttle commanded by the RC controller. So in order to have the computer take control, the computer's commanded throttle must be less than the RC throttle. Since we don't know what that would be, safest bet is to have the RC command full throttle to ensure that the computer autopilot will take control.

Reply:
Oh, right. Yeah that makes sense.